pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -223,7 +223,13 @@ def _log_gpus_metrics(self) -> None:
                 self.trainer.lightning_module.log(key, mem, prog_bar=False, logger=True)
             else:
                 gpu_id = int(key.split("/")[0].split(":")[1])
-                if gpu_id in self.trainer._accelerator_connector.parallel_device_ids:
+                parallel_device_ids = []
+                if isinstance(self.trainer.accelerator, GPUAccelerator):
+                    if isinstance(self.trainer.strategy, ParallelStrategy):
+                        parallel_device_ids = [i for i in range(len(self.trainer.strategy.parallel_devices))]
+                    elif isinstance(self.strategy, SingleDeviceStrategy):
+                        parallel_device_ids = [0]
```

Comment:
trainer.data_parallel_device_ids is the same logic as this. Let's use `trainer.data_parallel_device_ids`. 
This function is already deprecated, can be removed after 1.6 release.

Reply:
Sounds good